ferrocene/doc/internal-procedures/src/testing-other-targets.rst

```diff
@@ -0,0 +1,218 @@
+
+:ref:`aarch64-unknown-none`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. Note::
+    
+    In a :target:`aarch64-unknown-linux-gnu` environment (such as a Lima guest on :ref:`aarch64-apple-darwin`)
+    you may simply skip to the final step, running the tests, as no QEMU is needed.
+
+Install the necessary packages:
+
+.. code-block:: bash
+
+    sudo apt install gcc-aarch64-linux-gnu qemu-system-aarch64
+
+If you don't already have a ``/usr/share/binfmts/qemu-aarch64`` file, create one:
+
+.. code-block:: bash
+
+    package qemu-aarch64
+    interpreter /usr/bin/qemu-aarch64-static
+    magic \x7fELF\x02\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\xb7\x00
+    mask \xff\xff\xff\xff\xff\xff\xff\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff
+    credentials no
+    preserve no
+    fix_binary no
+    
+Then make sure it's imported:
+
+.. code-block:: bash
+    
+   sudo update-binfmts --import qemu-aarch64
+
+You can now run the tests:
+
+.. code-block:: bash
+
+    ./x test --stage 1 --target aarch64-unknown-ferrocenecoretest library/core
+
+:target:`thumbv7em-none-eabihf` & :target:`thumbv7em-none-eabi`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Install the necessary packages:
+
+.. code-block:: bash
+
+    sudo apt install gcc-arm-none-eabi qemu-system-arm
+
+If you don't already have a ``/usr/share/binfmts/qemu-arm`` file, create one:
+
+.. code-block:: bash
+
+    package qemu-arm
+    interpreter /usr/bin/qemu-arm-static
+    magic \x7fELF\x01\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x00\x28\x00
+    mask \xff\xff\xff\xff\xff\xff\xff\x00\xff\xff\xff\xff\xff\xff\xff\xff\xfe\xff\xff\xff
+    credentials no
+    preserve no
+    fix_binary no
+    
+Then make sure it's imported:
+
+.. code-block:: bash
+    
+   sudo update-binfmts --import qemu-arm
+
+.. note::
+    
+    Currently, this target uses our *secret sauce*. Eventually this will be an open source component of Ferrocene, but for now, it's our little bit of arcane magic.
+
+Now set the target:
+
+.. code-block:: bash
+
+    export TARGET="thumbv7em-ferrocenecoretest-eabihf"
+    # or 
+    export TARGET="thumbv7em-ferrocenecoretest-eabi"
+
+Next, it's possible to build the *secret sauce*, or to download it. Generally, it's easier to download it, but if necessary you can find the repository in the `Ferrocene <https://github.com/ferrocene>`_ organization.
+
+Refer to the ``.circleci/workflows.yml`` file on the ``setup-secret-sauce`` command to see which date/hash of the sauce to download.
+
+.. code-block:: bash
+
+    export SAUCE_DATE=20250121
+    export SAUCE_HASH=1671dac
+    
+    mkdir -p /tmp/ferrocene/$TARGET
+    aws s3 cp s3://ferrocene-ci-mirrors/coretest-secret-sauce/$SAUCE_DATE/$SAUCE_HASH/$TARGET.tar.gz /tmp/ferrocene/
+    tar xf /tmp/ferrocene/$TARGET.tar.gz --directory=/tmp/ferrocene/$TARGET
+
+You can now run the tests:
+
+.. code-block:: bash
+
+    export QEMU_CPU=cortex-m4
+    ./x test --stage 1 --target $TARGET library/core
```

Comment:
Does this command require downloading LLVM and in turn AWS login? I see a message to either add `download-ci-llvm = false` or `aws sso login`. I assumed that once I have the secret sauce package from above it should be ok.

Reply:
In general AWS access is required for us when building Ferrocene.